src/services.h

```diff
@@ -529,6 +529,13 @@ namespace dd
 	  pout._status = -3;
 	  throw;
 	}
+	  catch (const std::exception &e)
```

Comment:
OK this adds more in-depth granularity to error messages. As a rule-of-thumb, separate PRs when you are modifying core fles. Will merge this one as is though.